packages/features/vault/src/vault-view/vault-view.tsx

```diff
@@ -1,17 +1,35 @@
+import { useEffect } from 'react';
 import { PageLoading, SideBarLayout } from '@notional-finance/mui';
 import { useContext } from 'react';
-import { VAULT_SUB_NAV_ACTIONS } from '@notional-finance/shared-config';
+import { VAULT_ACTIONS } from '@notional-finance/shared-config';
 import { useVaultTransaction } from '../hooks/use-vault-transaction';
-import { VaultActionContext } from '../managers';
-// import { VaultAction } from './vault-action';
+import { VaultActionContext } from './vault-action-provider';
 import { VaultActionSideDrawer } from './vault-action-side-drawer';
 import { VaultSummary } from './vault-summary';
+import { useParams } from 'react-router';
+
+interface VaultParams {
+  vaultAddress?: string;
+  sideDrawerKey?: VAULT_ACTIONS;
+}
 
 export const VaultView = () => {
-  const { state } = useContext(VaultActionContext);
+  const { updateState } = useContext(VaultActionContext);
+  const { vaultAddress, sideDrawerKey } = useParams<VaultParams>();
   const txnData = useVaultTransaction();
   const showTransactionConfirmation = txnData ? true : false;
-  const { vaultAddress } = state || {};
+
+  useEffect(() => {
```

Comment:
Maybe you want to combine these two useEffect calls into one? I'm not sure if we will see race conditions if both action and address change at the same time. Probably better to be safe.